Remove support for Debian Buster

jsf9k opened the pull request:
## 🗣 Description ##

This pull request removes support for Debian Buster in all scenarios.

## 💭 Motivation and context ##

Buster's version of Python 3 is too old to support some of the tools that are installed, e.g., [maurosoria/dirsearch](https://github.com/maurosoria/dirsearch) which requires Python 3.8 or later.

Buster's Go and Rust support is also lacking, so it makes sense to just drop this antiquated platform.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Remove Debian Buster GH Actions checks as required.

jsf9k committed:
Remove support for Debian Buster in the python scenario

Buster's version of Python 3 is too old to support some of the tools
that are installed, e.g., maurosoria/dirsearch which requires Python
3.8 or later.

mcdonnnj reviewed (approved):
🫡

dav3r reviewed (approved):
👍

dv4harr10 commented:
Hi Team @all , I noticed a few occurrences of "buster" that maybe should be removed, Please advise: ansible-role-assessment-tool/molecule/default/externally-managed-python.yml , line 13 - Should "buster" be removed?

ansible-role-assessment-tool/tasks/create_python_venv.yml , line 16 - Should "buster" be removed?

ansible-role-assessment-tool/tasks/install_go.yml , lines: 7, 9, 12, 20, and 27. Should "buster" be removed?

jsf9k commented:
> Hi Team @ALL , I noticed a few occurrences of "buster" that maybe should be removed, Please advise: ansible-role-assessment-tool/molecule/default/externally-managed-python.yml , line 13 - Should "buster" be removed?

This language comes from upstream ([cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role)) and hence should not be changed here.

> ansible-role-assessment-tool/tasks/create_python_venv.yml , line 16 - Should "buster" be removed?

Yes.  ~I will take care of this.~  Please see commit 599d8b4.

> ansible-role-assessment-tool/tasks/install_go.yml , lines: 7, 9, 12, 20, and 27. Should "buster" be removed?

Yes.  ~I will take care of this.~  Please see commit 599d8b4.

jsf9k committed:
Remove support for Buster altogether

@dv4harr10 pointed out that I had left some Buster cruft lying around,
so this commit gets rid of that detritus.

Co-authored-by: David Harris <123905168+dv4harr10@users.noreply.github.com>
